File: JAVA/src/daysOfCode30/day19_Interfaces/Interfaces.java
package daysOfCode30.day19_Interfaces;

import java.util.Scanner;

interface AdvancedArithmetic{
   int divisorSum(int n);
}

//Write your code here
class Calculator implements AdvancedArithmetic{
	private int acum;
	public int divisorSum(int n){
		for(int i=1; i<=n; i++){
			if(n%i == 0)
				acum += i;
		}
		return acum;
	}
}


public class Interfaces {
		
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();
        
      	AdvancedArithmetic myCalculator = new Calculator(); 
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
        System.out.println(sum);
    }
}

File: PHP/30DaysOfCode/day19_Interfaces/Interfaces.php
<?php
interface AdvancedArithmetic{
    public function divisorSum($n);
}

//Write your code here
class Calculator implements AdvancedArithmetic{
    private $acum;
    public function divisorSum($n){
        for($i=1; $i<=$n; $i++){
            if($n%$i == 0)
                $acum += $i;
        }
        return $acum;
    }
}

$n=intval(fgets(STDIN));
$myCalculator=new Calculator();
if($myCalculator instanceof AdvancedArithmetic)//checking if Calculator has implemented AdvancedArithemtic
{
    $sum=$myCalculator->divisorSum($n);
    echo "I implemented: AdvancedArithmetic\n".$sum;
}
else
{
    echo "Wrong answer";// You will get this output if you dont implement
}
?>